aspen/request_processor/algorithm.py

```diff
@@ -70,6 +70,11 @@ def load_resource_from_filesystem(request_processor, dispatch_result):
     return {'resource': resources.get(request_processor, dispatch_result.match)}
 
 
-def render_resource(state, request_processor, resource):
-    state.setdefault('output', output.Output(charset=request_processor.charset_dynamic))
+def render_resource(state, resource):
     return {'output': resource.render(state)}
+
+
+def encode_output(output, request_processor):
+    if isinstance(output.body, unicode):
+        output.charset = request_processor.charset_dynamic
```

Comment:
We never decode static assets, so they're never `unicode`, but you're right that the names of the configuration knobs are bad, I considered changing them but couldn't decide on better ones.